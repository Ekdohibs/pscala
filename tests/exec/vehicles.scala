class Vehicle {
  var start = 10;
  var position = start;
  def move(d: Int) { position = position + d }
}

class Car extends Vehicle {
  var passengers = 0;

  def await(v: Vehicle) {
    if (v.position < position)
      v.move(position - v.position)
    else
      move(10)
  }
}

class Truck extends Vehicle {
  var load = 0;

  override def move(d: Int) {
    position = if (d <= 55) position + d else 55
  }
}

object Main {
  def main(args: Array[String]) {
    val t = new Truck();
    val c = new Car();
    c.passengers = 2;
    print(c.position); print("\n");
    c.move(60);
    print(c.position); print("\n");
    val v: Vehicle = c;
    v.move(70);
    print(c.position); print("\n");
    c.await(t);
    print(t.position); print("\n");
    print(c.position); print("\n")
  }
}

