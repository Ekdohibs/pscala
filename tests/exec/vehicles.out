10
70
140
55
140
